terraform/environments/core-shared-services/s3_malware_protection.tf

```diff
@@ -0,0 +1,29 @@
+# GuardDuty Malware Protection Plan
+resource "aws_guardduty_malware_protection_plan" "malware_protection_plan" {
+  for_each = toset([
+    "mod-platform-image-artefact-bucket20230203091453221500000001",
```

Comment:
Almost there! You're better off using the attributes of the Terraform resources than hardcoding their names as these would be subject to change.
`module.s3-bucket.bucket.id` for the image artefact bucket, and `module.s3-software-bucket.bucket.id` ought to be the correct references if I read the [module outputs](https://github.com/ministryofjustice/modernisation-platform-terraform-s3-bucket/blob/main/outputs.tf) correctly.